lesson20/main.js

```diff
@@ -1 +1,159 @@
+  }
+};
+
+const fetchUsersData = async () => {
+  try {
+    const json = await fetchDataInSecond(3000, jsonUrl);
+    return json.data;
+  } catch (e) {
+    tableWrap.textContent = "データの取得ができませんでした。";
+    console.error(e);
+  } finally {
+    loaded();
+  }
+};
+
+/**
+ * Rendering table using users data from JSON file
+ * @param {Object} usersData array of users data
+ */
+const renderTable = (usersData) => {
+  const table = createTable();
+  table.appendChild(createTableHeader(usersData));
+  table.appendChild(createTableData(usersData));
+  tableWrap.appendChild(table);
+};
+
+const createTable = () => {
+  const table = createElementWithClassName("table", "users-table");
+  table.id = "js-table";
+  return table;
+};
+
+/**
+ *Creating table header with key of usersData
+ * @param {Array} usersData The Array of usersData
+ */
+const createTableHeader = (usersData) => {
+  const trOfThead = document.createElement("tr");
+  const fragment = document.createDocumentFragment();
+  AddTextContentToThead(usersData, fragment);
+  trOfThead.appendChild(fragment);
+  return trOfThead;
+};
+
+/**
+ *Extract key from usersData
+ * @param {Object} usersData[0] value from one of userData in the object
+ * @param {String} key key of userData
+ */
+const AddTextContentToThead = (usersData, fragment) => {
+  Object.keys(usersData[0]).forEach((key) => {
+    const th = createElementWithClassName("th", "users-table__th");
+    th.textContent = formingTableHeaderNameWithKey(key);
+    fragment.appendChild(th);
+  });
+};
+
+const formingTableHeaderNameWithKey = (key) => {
+  switch (key) {
+    case "id":
+      return "ID";
+    case "name":
+      return "名前";
+    case "sex":
+      return "性別";
+    case "age":
+      return "年齢";
+    default:
+      console.error(`${key}は見つかりませんでした`);
+  }
+};
+
+/**
+ *Creating table data with value of usersData
+ * @param {Array} usersData The Array of usersData
+ */
+
+const createTableData = (usersData) => {
+  const fragment = document.createDocumentFragment();
+  usersData.forEach((userData) => {
+    const trOfTdata = document.createElement("tr");
+    /**
+     *Extract value from each of usersData
+     * @param {Object} userData the object of each users data
+     * @param {String,Number} val value of each users data
+     */
+
+    Object.values(userData).forEach((val) => {
+      const td = createElementWithClassName("td", "users-table__td");
+      td.textContent = val;
+      trOfTdata.appendChild(td);
+    });
+    fragment.appendChild(trOfTdata);
+  });
+  return fragment;
+};
+
+const init = async () => {
+  tableWrap.insertAdjacentElement("beforebegin", createLoader());
+  loading();
+  let usersData;
+  try {
+    usersData = await fetchUsersData();
+  } catch (e) {
+    console.error(e);
+  } finally {
+    console.log("処理が完了しました。");
+  }
+  if (usersData.length === 0) {
+    tableWrap.textContent = "データがありません。";
+    return;
+  } else {
+    renderTable(usersData);
+  }
+};
```

Comment:
IMO
The first one.

FROM
```js
  if (usersData.length === 0) {
    tableWrap.textContent = "データがありません。";
    return;
  } else {
    renderTable(usersData);
  }
```
TO
```js
  if (usersData.length === 0) {
    tableWrap.textContent = "データがありません。";
    return;
  }
  renderTable(usersData);
```
The second.
I think the handling of init() errors needs a little more usability.
<img width="1080" alt="スクリーンショット 2022-01-01 16 00 16" src="https://user-images.githubusercontent.com/43788628/147845581-23514d2e-cd90-4112-a2a1-83000a0245cd.png">
```suggestion
const init = async () => {
  tableWrap.insertAdjacentElement("beforebegin", createLoader());
  loading();
  let usersData;
  try {
    usersData = await fetchほげほげData(); 
  } catch (e) {
  //I feel that if we fix this, it will be even better.
    loaded();
    console.error(e);
    tableWrap.textContent = `error内容:${e.message}`;
  } finally {
    console.log("処理が完了しました。");
  }
  if (usersData.length === 0) {
    tableWrap.textContent = "データがありません。";
    return;
  }
  renderTable(usersData);
};
```

Reply:
- For the first one
I searched about quick return again and I learned it doesn't need `else`. I didn't understand about that.
Thank you for teaching me :)
[88e2fb9](https://github.com/yuka830/js-lessons/pull/26/commits/88e2fb986e6d3b0b3e4c0ce81f0ea593129c2ae1)

- For the second one
You are right, this improves the usability.
I implemented error messages are displayed in catch statement. I wrote error message in a similar way to other error messages.
And also I change calling `loaded()`　in finally statement not in `fetchJson`  but in `init`. I think finally statement is not must, so I deleted it in `fetchJson`.
What do you think? Thanks reviewing many times.
[bc308a9](https://github.com/yuka830/js-lessons/pull/26/commits/bc308a9af505a3c053396b3ddf4ac9ea258a491d)